packages/kbn-lens-embeddable-utils/config_builder/charts/gauge.ts

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import {
+  FormBasedPersistedState,
+  FormulaPublicApi,
+  GaugeVisualizationState,
+} from '@kbn/lens-plugin/public';
+import { DataView } from '@kbn/data-views-plugin/public';
+import { BuildDependencies, DEFAULT_LAYER_ID, LensAttributes, LensGaugeConfig } from '../types';
+import {
+  addLayerFormulaColumns,
+  buildDatasourceStates,
+  buildReferences,
+  getAdhocDataviews,
+} from '../utils';
+import { getFormulaColumn, getValueColumn } from '../columns';
+
+const ACCESSOR = 'metric_formula_accessor';
+
+function buildVisualizationState(config: LensGaugeConfig): GaugeVisualizationState {
+  if (config.layers.length !== 1) {
+    throw new Error('metric must define a single layer');
+  }
+
+  const layer = config.layers[0];
+
+  return {
+    layerId: DEFAULT_LAYER_ID,
+    layerType: 'data',
+    showBar: false,
```

Comment:
This is a metric Visualization property if I am not mistaken. Do we need to set it on gauges?